test/Benchmarks/Metrics/MetricCollectBenchmarks.cs

```diff
@@ -0,0 +1,110 @@
+// <copyright file="MetricCollectBenchmarks.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+
+/*
+BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19043
+Intel Core i7-8650U CPU 1.90GHz (Kaby Lake R), 1 CPU, 8 logical and 4 physical cores
+.NET Core SDK=5.0.400
+  [Host]     : .NET Core 3.1.18 (CoreCLR 4.700.21.35901, CoreFX 4.700.21.36305), X64 RyuJIT
+  DefaultJob : .NET Core 3.1.18 (CoreCLR 4.700.21.35901, CoreFX 4.700.21.36305), X64 RyuJIT
+
+
+|  Method | ExportDelta |      Mean |    Error |   StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
+|-------- |------------ |----------:|---------:|---------:|------:|------:|------:|----------:|
+| Collect |       False | 100.59 us | 1.970 us | 5.490 us |     - |     - |     - |     136 B |
+| Collect |        True |  98.41 us | 1.861 us | 4.670 us |     - |     - |     - |     136 B |
+*/
+
+namespace Benchmarks.Metrics
+{
+    [MemoryDiagnoser]
+    public class MetricCollectBenchmarks
+    {
+        private Counter<long> counter;
+        private MeterProvider provider;
+        private Meter meter;
+        private Random random = new Random();
```

Comment:
IIRC the benchmark tool is not running on multi-threads so there is no race condition?
It might be good idea to make this thread safe.